host_core/lib/host_core/providers/builtin/numbergen.ex

```diff
@@ -0,0 +1,17 @@
+defmodule HostCore.Providers.Builtin.Numbergen do
+  def invoke("GenerateGuid", _payload) do
+    Msgpax.pack!(UUID.uuid4())
+  end
+
+  def invoke("RandomInRange", payload) do
+    params = Msgpax.unpack!(payload)
+
+    min = min(params["min"], 0)
+    max = max(params["max"], 4_294_967_295)
+    Msgpax.pack!(Enum.random(min..max))
+  end
+
+  def invoke("Random32", _payload) do
+    Msgpax.pack!(floor(:rand.uniform(4_294_967_294)))
```

Comment:
What number does `4294967294` correspond to? Is it the largest 32-bit integer without a signed bit? If so, some languages use signed integers and may not support this big of a number if they try to store it in an int32

Reply:
It's the largest U32. The smithy spec for this value calls for it to be a u32 so I'm basically just enforcing that it can be no bigger than the biggest U32.